tests/job.go

```diff
@@ -10,37 +10,88 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"kubevirt.io/client-go/kubecli"
+)
 
-	. "github.com/onsi/gomega"
+const (
+	toSucceed = true
+	toFail    = false
 )
 
-func WaitForJobToSucceed(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() bool {
-		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Failed).NotTo(Equal(*job.Spec.BackoffLimit), "Job was expected to succeed but failed")
-		return job.Status.Succeeded > 0
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should succeed")
+// WaitForJobToSucceed blocks until the given job finishes.
+// On success, it returns with a nil error, on failure or timeout it returns with an error.
+func WaitForJobToSucceed(job *batchv1.Job, timeout time.Duration) error {
+	return waitForJob(job, toSucceed, timeout)
+}
 
+// WaitForJobToFail blocks until the given job finishes.
+// On failure, it returns with a nil error, on success or timeout it returns with an error.
+func WaitForJobToFail(job *batchv1.Job, timeout time.Duration) error {
+	return waitForJob(job, toFail, timeout)
 }
 
-func WaitForJobToFail(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() int32 {
-		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Succeeded).ToNot(BeNumerically(">", 0), "Job should not succeed")
-		return job.Status.Failed
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeNumerically(">", 0), "Job should fail")
+func waitForJob(job *batchv1.Job, toSucceed bool, timeout time.Duration) error {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return err
+	}
+
+	jobCompleteError := func(job *batchv1.Job) error { return nil }
+	jobFailedError := func(job *batchv1.Job) error { return nil }
+	if toSucceed {
+		jobFailedError = func(job *batchv1.Job) error {
+			return fmt.Errorf("Job %s finished with failure, status: %+v", job.Name, job.Status)
+		}
+	} else {
+		jobCompleteError = func(job *batchv1.Job) error {
+			return fmt.Errorf("Job %s finished with success, status: %+v", job.Name, job.Status)
+		}
+	}
+
+	for seconds := 0.0; seconds < timeout.Seconds(); seconds++ {
```

Comment:
This will not work as expected. It can happen that this will run for `timeout * 10` seconds.

Reply:
Sorry, I do not understand. Why did you multiply by 10?

- For timeout=0, it will not run.
- For timeout=1, it will run once.
- For timeout=2, it will run twice.
etc...

Reply:
But that does not mean it will run for 0,1,2 seconds... As call to Kubernetes API can block for some time...

Reply:
K8S API calls are asynchronous, they may take a few msec in the worse case, but not more than that.
Using a monotonic clock for such a inaccuracy is an overkill IMO, especially for the tests. I think we can live with a few msec drift here.